library/core/src/net/ip_addr.rs

```diff
@@ -67,12 +68,20 @@ pub enum IpAddr {
 /// assert!("0000000.0.0.0".parse::<Ipv4Addr>().is_err()); // first octet is a zero in octal
 /// assert!("0xcb.0x0.0x71.0x00".parse::<Ipv4Addr>().is_err()); // all octets are in hex
 /// ```
-#[derive(Copy, Clone, PartialEq, Eq, Hash)]
+#[derive(Copy, Clone, PartialEq, Eq)]
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct Ipv4Addr {
     octets: [u8; 4],
 }
 
+impl Hash for Ipv4Addr {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        // Hashers are often more efficient at hashing a fixed-width integer
+        // than a bytestring, so convert before hashing.
+        u32::from_le_bytes(self.octets).hash(state);
```

Comment:
You should probably note that you're not using `.to_bits()` here because that would require a byte-swap and we want to bias towards little-endian platforms. Or if it's fine for the hash to vary across platforms, use `from_ne_bytes`.

Reply:
I don't know if we can decide for users that it doesn't matter across platforms. The current implementation behaves the same on all platforms.

Reply:
The current documentation of `Hash` already says that the hash values [aren't portable](https://doc.rust-lang.org/std/hash/trait.Hash.html#portability).

Reply:
We can decide that, see [the portability section in the `Hash` documentation](https://doc.rust-lang.org/nightly/std/hash/trait.Hash.html#portability).